src/CMakeLists.txt

```diff
@@ -324,6 +324,28 @@ add_custom_target(hotgym
                   COMMENT "Executing ${src_executable_hotgym}"
                   VERBATIM)
 
+## Setup benchmark_hotgym for memory leak detection (with valgrind, debug mode) 
+#
+set(src_executable_memleak_hotgym memleak_hotgym)
+add_executable(${src_executable_memleak_hotgym} examples/Hotgym.cpp)
+target_link_libraries(${src_executable_memleak_hotgym}
+                ${core_library}
+                ${COMMON_OS_LIBS}
+                )
+target_compile_options( ${src_executable_memleak_hotgym} PUBLIC ${INTERNAL_CXX_FLAGS}) # add debug flag, required for valgrind
+target_compile_definitions(${src_executable_memleak_hotgym} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
+target_include_directories(${src_executable_memleak_hotgym} PRIVATE
+                ${CORE_LIB_INCLUDES}
+                ${EXTERNAL_INCLUDES}
+                )
+add_custom_target(test_memleak_hotgym
+	          -DMAKE_BUILD_TYPE:STRING=Debug
```

Comment:
how do I make a target with custom flags (here Debug mode, or `-g`, my problem is NDEBUG is still defined (it sets size of test-run the HelloSPTP test, and I need it very short for valgrind) 
@dkeeney ?

Reply:
ok, not necessarily needed, I hacked it with hotgym taking a command line param of number of runs. Valgrind can run in release mode, only the reported line-numbers will not match. 
Advantage of the current approach is that we use just 1 target: hotgym_example

Reply:
CMake controls what happens in debug vs release mode.  I came to the conclusion that it must be all one or all the other, no mix.  In the MSVC IDE you can do a build in Debug and then do a build in Release and it builds separate targets (for the most part). But for Linux and OSx I think you have to wipe the build folder to clear caches in order to switch between Release and Debug builds.  I tried mixing them...like having release dependency library while debugging and had a lot of problems getting CMake to do it right.  Perhaps I just did not have the right combinations.

Reply:
so I'll just leave it as is, valgrind can work in Release, just doesn't report useful line-numbers -> user will have to rebuild in Debug to find out what causes the fault, if mem leak is detected.